### PR TITLE
A kto kurwa mówił, że będzie lekko, nikt kurwa, i słowo się rzekło

### PR DESCRIPTION
**Pull request by kornelu123**

### CODE REVIEW

File: tea.cpp [hunk 1]
```diff
@@ -6,27 +6,27 @@
...
   this->key[1] = get_next(0,-1);
   this->key[2] = get_next(0,-1);
   this->key[3] = get_next(0,-1);
 
   const uint32_t u32_datasize = 
-    u8_datasize % TEA_DATABLOCK_SIZE == 0 ? 
-    u8_datasize/TEA_DATABLOCK_SIZE : 
-    (u8_datasize/TEA_DATABLOCK_SIZE + 1)*2;
+    (*u8_datasize) % TEA_DATABLOCK_SIZE == 0 ? 
+    ((*u8_datasize)/TEA_DATABLOCK_SIZE) * 2 :
```
Comment (kornelu123) [inline tea.cpp:18]: Here was a bug, fix this later

Comment (kornelu123) [reply tea.cpp:18]: i mean, was the bug, rebase this and add in another commit

File: ble.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  this->data_size = data_size;
+  this->data = (uint8_t *)std::malloc(data_size);
+  std::memcpy(this->data, data, data_size);
+}
+
+ble::packet::packet(const ble::packet *copy)
+{
+  this->data_size = copy->data_size;
+  this->data = (uint8_t *)std::malloc(copy->data_size);
+  std::memcpy(this->data, copy->data, copy->data_size);
```
Comment (kornelu123) [inline ble.cpp:23]: Either use `std::memcpy` or `memcpy` you moron.

**Review by kornelu123 — COMMENTED** [review]
:1st_place_medal:

**Comment by kornelu123** [comment]
reword `[ed9ffca](https://github.com/husxk/pwr-niduc-p/pull/9/commits/ed9ffca05e3750c88144f43bb3ded567e6dc6b87)`

**Comment by husxk** [comment]
I will close this PR. Please look at https://github.com/husxk/pwr-niduc-p/pull/11
I have changed some code here.

**Pull request closed without merging**